variable/comparison.cpp

```diff
@@ -15,11 +16,36 @@ namespace scipp::variable {
 
 namespace {
 Variable _values(Variable &&in) { return in.hasVariances() ? values(in) : in; }
+
+Variable tolerance(const Variable &atol, const Variable &rtol,
+                   const Variable &y) {
+  return atol + rtol * abs(y);
+}
 } // namespace
 
+bool allclose(const Variable &a, const Variable &b, const Variable &rtol,
+              const Variable &atol, const NanComparisons equal_nans) {
+  auto tol = tolerance(atol, rtol, b);
+  if (a.hasVariances() && b.hasVariances()) {
+    return allclose(values(a), values(b), rtol, atol, equal_nans) &
+           allclose(stddevs(a), stddevs(b), rtol, atol, equal_nans);
+  } else {
+    Variable result = makeVariable<bool>(Values{true});
+    if (equal_nans == NanComparisons::Equal)
+      accumulate_in_place(result, const_cast<Variable &>(a), b,
```

Comment:
Should figure a way around these `const_casts` too. Maybe just a typo on in the overload for `accumulate_in_place` that can be corrected.

Reply:
Unfortunately it can't: the mutability is used in `bin_detail.h` so we would need a different function.